files/image_config/config-setup/config-setup

```diff
@@ -99,28 +103,119 @@ run_hookdir() {
     return $exit_status
 }
 
+initalize_configdb()
+{
+    NAMESPACE_ARGUMENT=''
+    IP_NETNS_CMD=''
+    # Create correct argument based on per-asic name space in multi-npu platforms
+    if [[ $# -eq 2 ]]; then
+        NAMESPACE_ARGUMENT=${NAMESPACE_OPTION}${NAMESPACE_PREFIX}$2
+        IP_NETNS_CMD=${IP_NETNS_CMD_PREFIX}${NAMESPACE_PREFIX}$2
+    fi 
+    
+    if [ "$1" == "init" ]; then
+        # case when config db need to be created/initialized from minigraph
+        sonic-db-cli ${NAMESPACE_ARGUMENT} CONFIG_DB FLUSHDB
+        sonic-cfggen -H -m -j ${INIT_CFG_JSON} ${NAMESPACE_ARGUMENT} --write-to-db
+        sonic-db-cli ${NAMESPACE_ARGUMENT} CONFIG_DB SET "CONFIG_DB_INITIALIZED" "1"
+        sonic-cfggen ${NAMESPACE_ARGUMENT} -d --print-data > ${CONFIG_DB_PATH}${CONFIG_DB_PREFIX}$2${CONFIG_DB_SUFFIX}
+        if [[ ($NUM_ASIC -gt 1) && ($# -eq 1) ]]; then
+            return 0
+        fi
+        # this will be run only in per-asic namespace only in multi-npu platforms
+        ${IP_NETNS_CMD} config qos reload
+        ${IP_NETNS_CMD} pfcwd start_default
+    elif [ "$1" == "reload" ]; then
+        # case when config db is already present from previous image file system
+        sonic-cfggen ${NAMESPACE_ARGUMENT} -j ${CONFIG_DB_PATH}${CONFIG_DB_PREFIX}$2${CONFIG_DB_SUFFIX} --write-to-db
+    fi
+
+    return 0
+}
+configdb_migrator()
+{
+    NAMESPACE_ARGUMENT=''
+    IP_NETNS_CMD=''
+    # Create correct argument based on per-asic name space in multi-npu platforms
+    if [[ $# -eq 2 ]]; then
+        NAMESPACE_ARGUMENT=${NAMESPACE_OPTION}${NAMESPACE_PREFIX}$2
+    fi 
+
+    if [ "$1" == "init" ]; then
+        # case when config db created/initialized from minigraph need to be migrated
+        if [[ -x /usr/bin/db_migrator.py ]]; then
+            # Set latest version number
+            /usr/bin/db_migrator.py ${NAMESPACE_ARGUMENT} -o set_version
+        fi
+    elif [ "$1" == "reload" ]; then
+        # case when config db is already present from previous image file system
+        # need to be migrated
+        if [[ -x /usr/bin/db_migrator.py ]]; then
+            # Migrate the DB to the latest schema version if needed
+            /usr/bin/db_migrator.py ${NAMESPACE_ARGUMENT} -o migrate
+        fi
+    fi
+}
+
 # Reload minigraph.xml file on disk
 reload_minigraph()
```

Comment:
There is a reload_minigraph() in the updategraph service in  files/image_config/updategraph/updategraph. 
That needs to be modified in a similar way ?

Reply:
@judyjoseph for multi-npu platforms we are not supporting getting minigraph from DHCP service and doing reload as of now. That will be in TO-DO list

Reply:
@judyjoseph updated updategraph also